### PR TITLE
feat(NgbModalWindow): add NgbModalWindow component

### PR DESCRIPTION
**Pull request by pkozlowski-opensource**
NgbModalWindow is needed for modals support (Modal service).

### CODE REVIEW

File: src/modal/modal_window.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    if (this.keyboard) {
+      this.dismiss(ModalDismissReasons.ESC);
+    }
+  }
+
+  close(result: any) { this.closeEvent.emit(result); }
+
+  dismiss(reason) { this.dismissEvent.emit(reason); }
+
+  stopPropagation($event) { $event.stopPropagation(); }
```
Comment (wesleycho) [inline src/modal/modal_window.ts]: Can we type the event, or is that a TS limitation?

File: src/modal/modal_window.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  @Input() keyboard = true;
+  @Input() size: string;
+
+  @Output('close') closeEvent = new EventEmitter();
+  @Output('dismiss') dismissEvent = new EventEmitter();
+
+  constructor(private _elRef: ElementRef, private _dcl: DynamicComponentLoader) {}
+
+  @HostListener('click')
+  backdropClick() {
```
Comment (wesleycho) [inline src/modal/modal_window.ts]: I'm a bit confused by this - on click, it will run this function? Seems to be a bit of confusing naming here.

Comment (pkozlowski-opensource) [inline src/modal/modal_window.ts]: Which name is confusing? `backdropClick`? Can you think of an alternative proposal? `backdropClickHandler`?

Comment (Foxandxss) [inline src/modal/modal_window.ts]: It is fine for me.

File: src/modal/modal_window.ts (hunk 1 above)
Comment (Foxandxss) [inline src/modal/modal_window.ts]: IIRC you can type it as MouseEvent or something like that.

Comment (pkozlowski-opensource) [inline src/modal/modal_window.ts]: Done

**Comment by wesleycho** [comment]
LGTM

**Pull request closed without merging**